coilsnake/model/common/ips.py

```diff
@@ -74,5 +75,47 @@ def is_applied(self, rom):
                     return False
         return True
 
-    def create(self, clean_rom, modified_rom):
-        raise NotImplementedError("IPS patch creation is not implemented yet.")
+    def create(self, clean_rom, hacked_rom, patch_path):
```

Comment:
does this work even if the roms are different sizes?

Reply:
Here are my results with patching using an expanded 6MB ROM as the base one:

1. Base ROM Vanilla EB expanded to 6MB
2. Target ROM is MaternalBound (4MB)
3. Patch made with coilsnake-cli
4. Patch applied to a vanilla EB ROM expanded to 6MB (Blackscreen)
5. Patch applied to a vanilla EB ROM expanded to 4MB (Success)

Worth noting that the Expanded ROM used to make the MaternalBound ROM in CoiSnake was 4MB, not 6MB. I just grabbed a 6MB ROM as the base for testing how it behaves with the IPS made with this PR.

Reply:
As of this date, I have finally written (and tested) the code to prevent any cases where the clean ROM is greater in actual length than the modified ROM. The cases which I have tested all work like they should and are as follows:

1. Clean ROM with size 6 MB (48 MBits) and Modified ROM with size 4 MB (32 MBits).
2. Clean ROM with size 6 MB (48 MBits) and Modified ROM with size 3 MB (24 MBits).
3. Clean RON with size 4 MB (32 MBits) and Modified ROM with size 3 MB (24 MBits).

In written form more pleasing to the eye and easier to read:

(Clean ROM size - Modified ROM size)
1. 6 - 4
2. 6 - 3
3. 4 - 3

Unrelated, but I have also made it so that entering the metadata into the CLI createpatch command is not required. Hence, if you were to leave out the arguments for author, description and title, those values would be detected as None and re-assigned values like so: author='author', description='description', title='title'. This makes typing out the command a little bit easier if one doesn't care what any of those values are.